core/src/helpers.rs

```diff
@@ -169,16 +172,281 @@ pub fn starknet_block_to_eth_block(block: MaybePendingStarknetBlock) -> RichBloc
+                        .map(from_starknet_transaction_ethers_transaction)
+                        .collect(),
+                );
+                let header = Header {
+                    hash: Some(hash),
+                    parent_hash,
+                    uncles_hash: parent_hash,
+                    author: sequencer,
+                    miner: sequencer,
+                    state_root,
+                    // BlockWithTxHashes doesn't have a transactions root
+                    transactions_root: H256::zero(),
+                    // BlockWithTxHashes doesn't have a receipts root
+                    receipts_root: H256::zero(),
+                    number: Some(number),
+                    gas_used,
+                    gas_limit,
+                    extra_data,
+                    logs_bloom,
+                    timestamp,
+                    difficulty,
+                    nonce,
+                    size,
+                };
+                let block = Block {
+                    header,
+                    total_difficulty,
+                    uncles: vec![],
+                    transactions,
+                    base_fee_per_gas: None,
+                    size,
+                };
+                Rich::<Block> {
+                    inner: block,
+                    extra_info: BTreeMap::default(),
+                }
             }
         },
     }
 }
 
+fn from_starknet_transaction_ethers_transaction(tx: StarknetTransaction) -> EtherTransaction {
+    let mut ether_tx = EtherTransaction::default();
+
+    match tx {
+        StarknetTransaction::Invoke(invoke_tx) => {
+            match invoke_tx {
+                InvokeTransaction::V0(v0) => {
+                    // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+                    ether_tx.hash = H256::from_slice(&v0.transaction_hash.to_bytes_be());
+                    ether_tx.nonce = field_element_to_u256(v0.nonce);
+                    ether_tx.from = starknet_address_to_ethereum_address(v0.contract_address);
+                    // Define gas_price data
+                    ether_tx.gas_price = None;
+                    // Extracting the signature
+                    ether_tx.r = field_element_to_u256(v0.signature[0]);
+                    ether_tx.s = field_element_to_u256(v0.signature[1]);
+                    ether_tx.v = field_element_to_u256(v0.signature[2]);
+                    // Extracting the data (transform from calldata)
+                    ether_tx.input = vec_felt_to_bytes(v0.calldata);
+                    // How to fetch to?
+                    ether_tx.to = None;
+                    // How to fetch the value?
+                    ether_tx.value = U256::from(0);
+                    // How to fetch gas?
+                    ether_tx.gas = U256::from(0);
+                    // Extracting the chain_id
+                    ether_tx.chain_id = None;
```

Comment:
chain_id can be a constant. in fact it can be a workspace-level constant that is used in eth_chainId and here! you can perform a quick refactor if you'd like

Reply:
Ok!